boards/BMS/Src/App/states/App_PreChargeState.c

```diff
@@ -62,20 +62,24 @@ static void PreChargeStateRunOnTick100Hz(struct StateMachine *const state_machin
             precharge_relay, is_charger_connected, is_ts_rising_slowly, is_ts_rising_quickly,
             &precharge_fault_limit_exceeded);
 
-        struct HeartbeatMonitor *hb_monitor = App_BmsWorld_GetHeartbeatMonitor(world);
-        const bool               missing_hb = !App_SharedHeartbeatMonitor_Tick(hb_monitor);
-        App_CanTx_BMS_Warnings_MissingHeartBeat_Set(missing_hb);
 
+        // If there is a pre-charge fault and there were no more than three previous pre-charge faults
+        // Go back to Init State, add one to the pre-charge failed counter and set the CAN charging message to false
+        // Else go to Fault State, reset the pre-charge failed counter and set the CAN charging message to false
         if (has_precharge_fault)
         {
             const struct State *next_state =
                 (precharge_fault_limit_exceeded) ? App_GetFaultState() : App_GetInitState();
+            App_CanRx_DEBUG_ChargingSwitch_StartCharging_Update(false);
+            if (next_state == App_GetFaultState())
+            {
+                App_PrechargeRelay_ResetFaultCounterVal(precharge_relay);
+            }
             App_SharedStateMachine_SetNextState(state_machine, next_state);
```

Comment:
This logic might be clearer like:
```c
if(precharge_fault_limit_exceeded)
{
    App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
    App_PrechargeRelay_ResetFaultCounterVal(precharge_relay);
}
else
{
    App_SharedStateMachine_SetNextState(state_machine, App_GetInitState());
}
```

I feel like setting the next state ptr and then `if`ing based on that is confusing